pixelot/src/app/left-sidebar/left-sidebar.component.ts

```diff
@@ -227,17 +227,39 @@ export class LeftSidebarComponent {
 
   }
 
-  openLayerDialog(): void {
+  openLayerDialog(type: string): void {
     const dialogRef = this.dialog.open(AddLayerDialog, {
       width: '500px',
     });
 
+    if (type == 'tilemap') {
+      dialogRef.componentInstance.path_required = true;
+    }
+
     dialogRef.afterClosed().subscribe(result => {
       if (!result)
         return;
+      
+      console.log(result);
+
+      let new_layer: engine.RenderLayer;
+
+      if (type == 'tilemap') {
+        if (result.path == '') {
+          this._snackBar.open('Please enter a path', 'Close', {
+            duration: 5000
+          });
+          return;
+        }
 
-      engine.Renderer.addLayer(new engine.SpriteLayer(), result, engine.SceneManager.currentScene);
-      this.sceneData.addLayer(this.scene.name, result);
+        new_layer = engine.TileMapJSONParser.parse(result.path);
```

Comment:
Maybe wrap this in a try-catch and display a snackbar like above if parsing fails